substrate/frame/revive/src/lib.rs

```diff
@@ -978,67 +1028,99 @@ pub mod pallet {
 			#[pallet::compact] storage_deposit_limit: BalanceOf<T>,
 			code: Vec<u8>,
 			data: Vec<u8>,
+			effective_gas_price: U256,
+			encoded_len: u32,
 		) -> DispatchResultWithPostInfo {
+			let origin = Self::ensure_eth_origin(origin)?;
+			let info = T::FeeInfo::dispatch_info(
+				&Call::<T>::eth_instantiate_with_code {
+					value,
+					gas_limit,
+					storage_deposit_limit,
+					code: code.clone(),
```

Comment:
It feels wasteful that we need to copy everything here. Is there no other way? Now that `eth_instantiate_with_code` and `eth_call` cannot be dispatched directly through the caller, we could add the dispatch info as an additional argument to `eth_instantiate_with_code`, which we add in `create_call`. The weight function of `eth_instantiate_with_code` would not depend on that argument, so we can make it optional, set it to `None` first, then create the dispatch info in `create_call` and then replace the `None` argument with it.

Probably premature optimization.